holoviews/core/data/image.py

```diff
@@ -0,0 +1,250 @@
+        The set of samples available along a particular dimension.
+        """
+        dim_idx = dataset.get_dimension_index(dim)
+        if dim_idx in [0, 1]:
+            l, b, r, t = dataset.bounds.lbrt()
+            dim2, dim1 = dataset.data.shape[:2]
+            d1_half_unit = float(r - l)/dim1/2.
+            d2_half_unit = float(t - b)/dim2/2.
+            d1lin = np.linspace(l+d1_half_unit, r-d1_half_unit, dim1)
+            d2lin = np.linspace(b+d2_half_unit, t-d2_half_unit, dim2)
+            if expanded:
+                values = np.meshgrid(d2lin, d1lin)[abs(dim_idx-1)]
+                return values.flatten() if flat else values
+            else:
+                return d2lin if dim_idx else d1lin
+        elif dataset.ndims <= dim_idx < len(dataset.dimensions()):
+            # Raster arrays are stored with different orientation
+            # than expanded column format, reorient before expanding
+            if dataset.data.ndim > 2:
+                data = dataset.data[:, :, dim_idx-dataset.ndims]
+            else:
+                data = dataset.data
+            data = np.flipud(data)
+            return data.T.flatten() if flat else data
+        else:
+            return None
+
+
+    @classmethod
+    def select(cls, dataset, selection_mask=None, **selection):
+        """
+        Slice the underlying numpy array in sheet coordinates.
+        """
+        selection = {k: slice(*sel) if isinstance(sel, tuple) else sel
+                     for k, sel in selection.items()}
+        coords = tuple(selection[kd.name] if kd.name in selection else slice(None)
+                       for kd in dataset.kdims)
+        if not any([isinstance(el, slice) for el in coords]):
+            return dataset.data[dataset.sheet2matrixidx(*coords)]
+
+        # Apply slices
+        xidx, yidx = coords
+        l, b, r, t = dataset.bounds.lbrt()
+        if isinstance(xidx, slice):
+            l = l if xidx.start is None else max(l, xidx.start)
+            r = r if xidx.stop is None else min(r, xidx.stop)
+        if isinstance(yidx, slice):
+            b = b if yidx.start is None else max(b, yidx.start)
+            t = t if yidx.stop is None else min(t, yidx.stop)
+        bounds = BoundingBox(points=((l, b), (r, t)))
+        slc = Slice(bounds, dataset)
+        data = slc.submatrix(dataset.data)
+        return data
+
+
+    @classmethod
+    def sample(cls, dataset, samples=[]):
+        """
+        Sample the Raster along one or both of its dimensions,
+        returning a reduced dimensionality type, which is either
+        a ItemTable, Curve or Scatter. If two dimension samples
+        and a new_xaxis is provided the sample will be the value
+        of the sampled unit indexed by the value in the new_xaxis
+        tuple.
+        """
+        if len(samples[0]) == 1:
+            return dataset.select(**{dataset.kdims[0].name:
+                                     [s[0] for s in samples]}).columns()
+        else:
```

Comment:
No real need for the else as you are returning from the other branch. I like this method, it is relatively simple. :-)